### PR TITLE
dk/login as organizer or attendee

### PR DESCRIPTION
**Pull request by kos1kov**
add role to database and sign up as organizer and attendee

### CODE REVIEW

File: db/migrate/20190813190631_add_role_column_to_user.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddRoleColumnToUser < ActiveRecord::Migration[5.2]
+  def change
+    add_column :users, :role, :integer, default: 0
+  end
+end
```
Comment (LuginaJulia) [inline db/migrate/20190813190631_add_role_column_to_user.rb]: eof

File: app/controllers/api/v1/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# rubocop:disable Style/ClassAndModuleChildren
+class Api::V1::UsersController < ApplicationController
+  before_action :authenticate_user!
+
+  def index
```
Comment (anatoliliotych) [inline app/controllers/api/v1/users_controller.rb]: where do you use it?

File: config/routes.rb [hunk 3]
```diff
@@ -1,7 +1,12 @@
 Rails.application.routes.draw do
   root 'welcome#index'
-  get '*page', to: 'static#index',
+  get '*page', to: 'welcome#index',
```
Comment (anatoliliotych) [inline config/routes.rb]: why this change was done?

File: app/controllers/api/v1/users_controller.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+    render json: current_user
```
Comment (ermaac) [inline app/controllers/api/v1/users_controller.rb]: index action usually returns array of something

File: config/routes.rb (hunk 3 above)
Comment (kos1kov) [reply config/routes.rb]: because we have welcome controller

File: app/controllers/api/v1/users_controller.rb (hunk 2 above)
Comment (kos1kov) [reply app/controllers/api/v1/users_controller.rb]: Used when tested, not used at now, delete?

File: app/javascript/packs/application.jsx [hunk 5]
```diff
@@ -40,7 +42,11 @@ const MainNavigation = () => (
   </nav>
 );
 
-const DefaultLayout = ({component: Component, ...rest}) => {
+const DefaultLayout = ({ component: Component, ...rest }) => {
+  let user_name = { first_name: '', last_name: '', role: '' };
```
Comment (ermaac) [inline app/javascript/packs/application.jsx]: camelCase

File: app/javascript/packs/application.jsx [hunk 6]
```diff
@@ -40,7 +42,11 @@ const MainNavigation = () => (
...
+  if (sessionStorage.user_attributes != null) {
+    user_name = JSON.parse(sessionStorage.user_attributes);
```
Comment (ermaac) [inline app/javascript/packs/application.jsx]: try 
const { firstName, lastName, role } = JSON.parse(sessionStorage.user_attributes);

File: app/javascript/packs/application.jsx [hunk 7]
```diff
@@ -50,24 +56,27 @@ const DefaultLayout = ({component: Component, ...rest}) => {
   return (
     <Route {...rest} render={matchProps => (
       <div className="DefaultLayout">
+        {user_name.first_name + ' '}
```
Comment (ermaac) [inline app/javascript/packs/application.jsx]: move this to const userInfo, use string interpolation for building string

File: app/javascript/packs/application.jsx [hunk 8]
```diff
@@ -50,24 +56,27 @@ const DefaultLayout = ({component: Component, ...rest}) => {
...
       </div>
-    )} />
+    )}/>
   )
 };
 
 const Main = () => (
   <Switch>
-    <DefaultLayout exact path="/" component={App} />
+    <DefaultLayout exact path="/" component={App}/>
```
Comment (ermaac) [inline app/javascript/packs/application.jsx]: this change doesn't match to style guide

File: app/javascript/components/SignOut.jsx [hunk 9]
```diff
@@ -1,31 +1,32 @@
...
       sessionStorage.removeItem('user');
+      sessionStorage.removeItem('user_attributes');
       this.props.history.push('/');
-    }).catch(error => {
-      sessionStorage.removeItem('user');
-      this.setState({hasError: true})
-    })
-  };
+    });
+
```
Comment (ermaac) [inline app/javascript/components/SignOut.jsx]: extra blank line

File: app/javascript/packs/application.jsx [hunk 10]
```diff
@@ -40,35 +42,43 @@ const MainNavigation = () => (
   </nav>
 );
 
-const DefaultLayout = ({component: Component, ...rest}) => {
+
+const DefaultLayout = ({ component: Component, ...rest }) => {
+  let userInfo = '';
+  if (sessionStorage.user_attributes != null) {
+    const { first_name, last_name, role }   = JSON.parse(sessionStorage.user_attributes);
```
Comment (ermaac) [inline app/javascript/packs/application.jsx]: extra space

File: app/javascript/packs/application.jsx [hunk 11]
```diff
@@ -40,35 +42,43 @@ const MainNavigation = () => (
   </nav>
 );
 
-const DefaultLayout = ({component: Component, ...rest}) => {
+
+const DefaultLayout = ({ component: Component, ...rest }) => {
+  let userInfo = '';
+  if (sessionStorage.user_attributes != null) {
```
Comment (ermaac) [inline app/javascript/packs/application.jsx]: user_attributes most likely can be undefined rather than null
